cpp_exercise/src/lib/filehandlerlib.cc

```diff
@@ -35,18 +34,15 @@ void FileHandler::writeFile(const std::vector<unsigned char> & file_input, size_
     int err;
     _fd = open(_file_name.c_str(), O_WRONLY | O_APPEND, _mode);
```

Comment:
Your are wasting some power by opening/closing every time you want to write in the file (each package). I recommend to open at the start and close at the end. (same comment for reading)

Reply:
It's not what I mean: Opening and closing a file cost processor resources, And you are doing it every time you want to write(or read) on the file.